### DISCUSSION TITLE
Better "Music Library" management - Handling "Disk" tag

### DISCUSSION THREAD

**Discussion by NexusPenguin** [rern/rAudio#1281] [Feature Requests]
Hi,

I really like rAudio and I use it on a daily basis. 

But to be able to use it, I use a third party remote controller. 
Simply but because the Music Library management is not elaborated enough in rAudio. 

There is a probably relatively "long" list of requested features to have here, but I think a really good start would be to consider the "Disk" tag. 

Simply put : it can happen that an album spreads over 2 disks. 
In this case, it can often happen that you have "Disk1" folder and "Disk 2" folder. 
In metadata, all tracks have same "Artist" and same "Album" tag. 
Still, if you browse the music library and search for artists, then album will be listed twice (or thrice if there are 3 disks). 

I would be super happy if the same album consisting of 2 disks could be listed only once...
But of course, when displaying the content, still being able to play in the correct order : 
Disk 1 Track 1 => Disk 1 Track 2 => Disk 1 track 3 => Disk 2 Track 1 => Disk 2 Track 2 => Disk 2 Track 3 => Disk 3...
And not what a lot of players do : 
Disk 1 Track 1 => Disk 2 Track 1 => Disk 3 Track 1... Which is of course super inconvenient for concept albums (i.e. Pink Floyd  "the Wall") or Opera or live albums...

Is this a feature that can be added ? 

Would be superb. 

Thanks a lot.

**Comment by R101G** [comment]
I don't see this problem. I have an Artist Folder/Album Folder/CD1 Folder + CD2 Folder. I use EasyTag to edit the metadata.

**Reply by NexusPenguin** [reply]
Hi,

So it you have CD1 and C2 folder, it is shown as a single album ? 

Me, rAudio shows 2 separate albums (or more). 

Here is an example. 

I have taken the complete cycle of Beethoven's Symphonies under the direction of Thomas DAUSGAARD. 
There are 6 disks for this cycle. Each disk has it's own name. 
But I retagged everyting to have the same "Album" name in meta data. 
I then created a different disk for each symphonie. 
So I can play them one after the other in the correct order. 

Here is how metadata looks like (shown in GMusicBrowser) : 
![DAUSGAARD-gmusic](https://user-images.githubusercontent.com/8678897/227772486-5d069cb8-1575-45b2-a142-590c0bdef1f2.png)

But if I want to listen to this album here is what is shown in "Albums" from rAudio. As you can see : several albums with no cover art by the way...
![DAUSGAARD-Album](https://user-images.githubusercontent.com/8678897/227773326-e7bbc043-7232-4b95-b8de-1e17aa78bdf9.png)

So I can decide to display the albums but going through "Album artist". Then here is what I get : 
![DAUSGAARD-ArtistAlbum](https://user-images.githubusercontent.com/8678897/227773398-3c00da04-3a78-4361-806e-3ec14165dbf0.png)

Still not superbly efficient :-(

If I try now to enter by "Artist" instead of "Album Artist", I get about the same result...

So maybe my tags are wrong, but as you can see, ALL TRACKS ARE UNDER THER SAME ALBUM NAME. 

Still they appear as separate albums, according to folders...

I think the tags are OK, because if I want to display those exact same folders under Emby, here is what appears : 
![DAUSGAARDEmby](https://user-images.githubusercontent.com/8678897/227773656-1b52983c-40a7-4291-bbbd-924712e84f8b.png)

All tracks under same album and each symphony under it's own Disk number... as it should be :-) Same if I display with Plex, or with BubbleUPnP or with Symfonium... used as "remote controls" for rAudio. 

So if you can tell me what I am doing wrong, I would be able to re-tag accordingly :-)

**Reply by R101G** [reply]
I think you probably need a folder with the album name under the Artist Folder. The Album Name folder should have a folder for each CD, plus duplicate cover art which appears in the library album icon. So, Artist > Album folder with jpg's + 6 x CD folders containing tracks and jpg's. Playing the album plays all the CD's, or go down another step to choose one. 

You might need to name the CD folders CD1 to CD6 or something, to indicate the order.

**Reply by NexusPenguin** [reply]
Hi,

Thanks, but that is EXACTLY what I have. 

All CD in the same folder and folder under artist's name 
So here I have : 
- BEETHOVEN => DAUSGAARD => CA + CD2 + CD3...

But I have exactly the same problem with this : 
BRUCE SPRINSTEEN => Live 1975-1985 => CD1 + CD2 + CD3...
They appear as 3 different CD's and not one Album (same goes for 2 other Live albums). 
As you can see on this screenshot...

![Springsteen](https://user-images.githubusercontent.com/8678897/227794253-60d27ae8-a6e3-4f9d-b2c3-770bcdfac48c.png)

As I said, Emby, Plex, Symfonium, BubbleUPnP can handle this with the same tags exactly as I expect it :-(

To make it more explicit : ifg I browser by artist, here is the thing : 

-Under "Bruce Springsteen" folder
![Springsteen2](https://user-images.githubusercontent.com/8678897/227794483-b11c5feb-2341-4f89-ac22-b5705f375814.png)

And then, in each of those "Live 1975-1985" folders, same content as pictured here : 
![Springsteen3](https://user-images.githubusercontent.com/8678897/227794509-4bfae35a-472e-40a8-82c2-91607c85e8e3.png)

I ma sorry to say it, but I do not think there's is something wrong with my files, but with the way they are handled by rAudio library management :-(
If there was something wrong with the way my files are sorted/metadata, almost all software would fail at sorting those would they not ?

**Reply by R101G** [reply]
What Library options have you enabled? Try disabling them all except your storage.

**Reply by NexusPenguin** [reply]
Hi,

OK. I'll try and get back to you :-)

Let's hope it helps :-)

Regards. 

DONE. All options disabled. Library updated "Completely". Exactly the same result. Nothing changed... Definitely a problem with rAudio from my POV :-(

**Comment by zedebugger** [comment]
@NexusPenguin 
It seems to me your issue is similar to mine that I raised unsuccessfully two years ago (https://github.com/rern/rAudio-1/discussions/118).
It is a shame raudio does not only interpret tags and build the library from there, but instead, like many other projects, also depends on a certain directory structure in representing a media library.

**Reply by NexusPenguin** [reply]
Hi,

Yes. And I also am unable to display Lyrics for some unexplainable reason. 
As Lyrics were the main reason for me staying on what seems to be an almost dying project, I am consideringt switching to something else... 
Regards.

**Comment by clriis** [comment]
I really don't understand how you get this problem:

The Kinks: Picture Book got 7 CD's. I renumbered the tracks 1-138 under one album tag and kept the CD tag (1-7):

![image](https://user-images.githubusercontent.com/16999072/233007880-4b8e92c5-2890-442b-a704-5ebdd62ca66e.png)

In rAudio they all show up neatly in ONE folder:

![image](https://user-images.githubusercontent.com/16999072/233008297-a2e8d0a8-3f93-48ec-85ea-d76722be716a.png)

![image](https://user-images.githubusercontent.com/16999072/233009058-303f2998-9079-49e7-ac93-ea5a4973e17e.png)

![image](https://user-images.githubusercontent.com/16999072/233010355-15a5a258-5b0d-47d6-a38a-bc5d67c3915c.png)

**Reply by NexusPenguin** [reply]
Hi,

Thanks for your reply. 
But sadly that does not adress anything. 
I consider it as just an uncomfortable workaround. 

Of course I could retag or rename or reorganize my albums and tracks for dozens of hours to have rAudio showing things right. 
But a lot of other apps do that with no problems. So I still think the problem is not my library, but the way it is handled by rAudio. 

No fixes to expect. 
And still no lyrics for unexplainable reason. 
So the switch to something else is close at hand. 

Regards.

**Reply by clriis** [reply]
I am curious.

Which other sound applications for Raspberry Pi is able to handle your request?

Regarding lyrics, I got no problem as well:

![image](https://user-images.githubusercontent.com/16999072/233272833-bac48b71-87df-4ce0-a698-532e1b2d0dd3.png)


Kind regards,

**Reply by NexusPenguin** [reply]
Hi, 

Thanks for your answer. 

I know. 
Lyrics are workiong for everybody, except me. 

So I probably have a problem with my network. 
I will have to try to solve this on my own. 
There is literally no one able to answer a question like : "Is a specific port used to retrieve lyrics ?". 

Regards.

**Comment by rern** [comment]
@NexusPenguin,

For the Album mode list, please try:
- Edit file: `/srv/http/mpdlibrary.php`
    - Replace line 53 - 92, `case 'find':` to `break;`, with:
    ```php
	case 'find':
		$format = str_replace( '%artist%', '[%albumartist%|%artist%]', $format );
		if ( is_array( $mode ) ) {
			exec( 'mpc find -f "'.$format.'" '.$mode[ 0 ].' "'.$string[ 0 ].'" '.$mode[ 1 ].' "'.$string[ 1 ].'" 2> /dev/null '
					."| awk 'NF && !a[$0]++'"
				, $lists );
			if ( ! count( $lists ) ) { // find with albumartist
				exec( 'mpc find -f "'.$format.'" '.$mode[ 0 ].' "'.$string[ 0 ].'" albumartist "'.$string[ 1 ].'" 2> /dev/null '
						."| awk 'NF && !a[$0]++'"
					, $lists );
			}
		} else {
			exec( 'mpc find -f "'.$format.'" '.$mode.' "'.$string.'" 2> /dev/null '
					."| sed 's:[^/]*$::'"
					."| awk 'NF && !a[$0]++'"
				, $lists );
		}
		if ( count( $f ) > 3 ) {
			htmlTrack( $lists, $f );
		} else { // modes - album, artist, albumartist, composer, conductor, date, genre
			htmlFind( $lists, $f );
		}
		break;
    ```
- See whether it fixed the issue.

**Reply by NexusPenguin** [reply]
Hi,

Many thanks to take time to answer and provide code to try to help me. 

I sincerely appreciate it... 

But I think I am not skilled enough to used rAudio as it should be. 
Why ? 
Because, whatever I do, I am unable to display lyrics. 
Besides, I think I do not understand how CamillaDSP works as I am unable to get something else than a 44100 capture samplerate. 
And of course, last but not least, this problem with the album sorting...

All in all, I think despite I love it, I am not familiar enough with some settings to use rAudi ot it's full extend. 

EDIT : 
For some unexplainable reason, I am now at least able to display lyrics that are contained in Metadata. 
But if no lyrics in metadata, it will not retrieve them from the internet as I think it should do :-(

EDIT 2 : 
Nope, lyrics gone again even if included in Metadata or in .txt file in the same folder as the played track...
